scripts/generate-readme.sc

```diff
@@ -0,0 +1,36 @@
+import $ivy.`org.scalameta::mdoc:2.1.5`
+import ammonite.ops._
+import java.nio.file.Paths
+import mdoc.Reporter
+import mdoc.StringModifier
+import scala.meta.inputs.Input
+
+class ListModifier() extends StringModifier {
+  override val name: String = "list"
+
+  override def process(
+      topic: String,
+      code: Input,
+      reporter: Reporter
+  ): String = {
+    val targetPaths = ls ! pwd/s"${topic}"/'resources
+    val targetNames = targetPaths.map(_.baseName).toList
+    targetNames.mkString(" - ", "\n - ", "\n")
+  }
+}
+
+val mdocSettings = mdoc
+  .MainSettings()
+  .withIn(Paths.get("temp"))
+  .withOut(Paths.get(""))
+  .withStringModifiers(List(new ListModifier))
+  .withNoLinkHygiene(true)
+
+// NOTE: (ckipp01) This is sort of a hack, but I don't know a different way to
+// this with mdoc having the template at the root and needed it to be renamed
+// and then put back into the root. This is a hack around that :/
+mkdir! pwd/'temp
+mv(pwd/"README.md.template", pwd/'temp/"README.md")
+mdoc.Main.process(mdocSettings)
+mv(pwd/'temp/"README.md", pwd/"README.md.template")
+rm! pwd/'temp
```

Comment:
😬 

It works...

Ha, so yea it's a bit hacky, but I don't know of another way to have the `.template` file in the root, have mdoc process it and have it turned into the `README.md` file back in the root.

@olafurpg this was the context of the question I asked you if you were curious.

Reply:
It's fine by me!

Reply:
Seeing this thread makes me wonder if we should extend —in to allow passing in a single file in which case that file gets written directly to —out (which also needs to be a file in that case) 🤔 would be easy to implement and more intuitive than writing path marcher globs or regexp

Reply:
I have several times wanted the mdoc cli to support a list of in/out pairs (for example for Docusaurus docs+blog)

Reply:
Yea, I was actually just looking at this:
https://github.com/scalameta/mdoc/blob/30dec31f2ba779cbe80a9ed9e43aab5ba06cf244/mdoc/src/main/scala/mdoc/internal/cli/MainOps.scala#L124

And was sort of thinking that a check could be done here to see if the `--in` is a file or directory, and if a file, then just allow for it to only process that file rather than treating is as a directory.

Reply:
That sounds like a good solution 👍